app/utils/chart-config/versioning.ts

```diff
@@ -1856,6 +1856,51 @@ export const configuratorStateMigrations: Migration[] = [
 
       return newConfig;
     },
+    down: (config) => {
+      const oldConfig = {
+        ...config,
+        version: "4.0.0",
+      };
+
+      if (!Array.isArray(oldConfig.chartConfigs)) {
+        return oldConfig;
+      }
+
+      oldConfig.chartConfigs = oldConfig.chartConfigs.map(
+        (chartConfig: any) => {
+          const revertedConfig = { ...chartConfig };
+
+          if (!isNotTableOrMap(revertedConfig)) {
```

Comment:
Nit: I think it a bit easier to reason about if we check for "true" conditions, maybe we could have `isTableOrMap` util and negate it here? Essentially the current condition means `isTableOrMap`, but is written a bit in reverse 😅 

Here generally I'd also try to avoid using imported functions, as they can be modified at any moment and break the migrations unknowingly. It would be safer to actually have `revertedConfig.chartType !== "map" && revertedConfig.chartType !== "table"` 👍